src/components/TrainCard.tsx

```
@@ -154,11 +164,90 @@ export default function TrainCard({
+		// Cleanup old entries
+		const now = Date.now();
+		const MAX_AGE_MS = 60 * 60 * 1000; // 1 hour
+		const MAX_ENTRIES = 1000;
+
+		// Remove entries older than 24 hours
+		Object.keys(trackMemory).forEach(trainNumber => {
+			const entry = trackMemory[trainNumber];
+			if (now - entry.timestamp > MAX_AGE_MS) {
+				delete trackMemory[trainNumber];
+			}
+		});
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)